src/main/java/com/minecolonies/coremod/colony/ColonyList.java

```diff
@@ -0,0 +1,256 @@
+package com.minecolonies.coremod.colony;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.minecolonies.coremod.util.Log;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+//TODO extend list/collection
+/**
+ * Data structure for storing colonies, optimized for performance.
+ *
+ * @param <T> Type of IColony (Colony or ColonyView)
+ *
+ * @author Colton
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') At-clauses have to appear in the order '[@author, @version, @param, @return, @throws, @exception, @see, @since, @serial, @serialField, @serialData, @deprecated]'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.javadoc.AtclauseOrderCheck)